File: Trainer.cs
using System;
using System.Collections.Generic;

public class Trainer
{
    public string Name { get; private set; }
    private List<Pokeball> belt;

    public Trainer(string name)
    {
        Name = name;
        belt = new List<Pokeball>();

        // Initialiseer de riem met zes Pokeballs, elk met een Charmander erin.
        for (int i = 0; i < 6; i++)
        {
            Charmander charmander = new Charmander();
            Pokeball pokeball = new Pokeball();
            pokeball.Contain(charmander);
            belt.Add(pokeball);
        }
    }

    public void ThrowPokeball(int pokeballNumber)
    {
        if (pokeballNumber < 1 || pokeballNumber > belt.Count)
        {
            Console.WriteLine("Invalid Pokeball number.");
            return;
        }

        // Vermindert met 1 omdat de lijstindexen bij 0 beginnen
        belt[pokeballNumber - 1].Throw();
    }

    public void ReturnPokemonToPokeball(Charmander charmander, int pokeballNumber)
    {
        if (pokeballNumber < 1 || pokeballNumber > belt.Count)
        {
            Console.WriteLine("Invalid Pokeball number.");
            return;
        }

        // Vermindert met 1 omdat de lijstindexen bij 0 beginnen
        belt[pokeballNumber - 1].Return(charmander);
    }

    // Optioneel: Methode om extra validatie toe te voegen bij het toevoegen van een Pokeball.
    public void AddPokeballToBelt(Pokeball pokeball)
    {
        if (belt.Count >= 6)
        {
            throw new InvalidOperationException("Cannot add more than six Pokeballs to the belt.");
        }
        belt.Add(pokeball);
    }
}

File: Opdracht 2/trainer.py
# trainer.py
from pokeball import Pokeball

class Trainer:
    def __init__(self, name):
        self.name = name
        self.belt = [Pokeball() for _ in range(6)]
        for pokeball in self.belt:
            pokeball.retrieve_charmander()  # Load each Pokeball with a Charmander

    def throw_pokeball(self):
        if self.belt:
            pokeball = self.belt.pop(0)
            pokeball.throw()
            return pokeball
        else:
            print(f"{self.name} has no more Pokeballs!")

    def return_pokemon(self, pokeball):
        if len(self.belt) < 6:
            pokeball.retrieve_charmander()
            self.belt.append(pokeball)
        else:
            raise Exception("Cannot carry more than six Pokeballs on the belt.")

